service/dap/server_test.go

```diff
@@ -5744,3 +5743,311 @@ func TestBadlyFormattedMessageToServer(t *testing.T) {
 		client.ExpectDisconnectResponse(t)
 	})
 }
+
+func TestDisassemble(t *testing.T) {
+	runTest(t, "increment", func(client *daptest.Client, fixture protest.Fixture) {
+		runDebugSessionWithBPs(t, client, "launch",
+			// Launch
+			func() {
+				client.LaunchRequest("exec", fixture.Path, !stopOnEntry)
+			},
+			// Set breakpoints
+			fixture.Source, []int{17},
+			[]onBreakpoint{{
+				// Stop at line 17
+				execute: func() {
+					checkStop(t, client, 1, "main.main", 17)
+
+					client.StackTraceRequest(1, 0, 1)
+					st := client.ExpectStackTraceResponse(t)
+					if len(st.Body.StackFrames) < 1 {
+						t.Fatalf("\ngot  %#v\nwant len(stackframes) => 1", st)
+					}
+					// Request the single instruction that the program is stopped at.
```

Comment:
Should we test for 0 instructions as well?

Reply:
done.